lib/report_plugins/autotune/autotune-prep/categorize.js

```diff
@@ -0,0 +1,410 @@
+var tz = require('moment-timezone');
+var basal = require('../profile/basal');
+var getIOB = require('../iob');
+var ISF = require('../profile/isf');
+var find_insulin = require('../iob/history');
+var dosed = require('./dosed');
+
+// main function categorizeBGDatums. ;) categorize to ISF, CSF, or basals.
+
+function categorizeBGDatums(opts) {
+    var treatments = opts.treatments;
+    // this sorts the treatments collection in order.
+    treatments.sort(function (a, b) {
+        var aDate = new Date(tz(a.timestamp));
+        var bDate = new Date(tz(b.timestamp));
+        //console.error(aDate);
+        return bDate.getTime() - aDate.getTime();
+    });
+    var profileData = opts.profile;
+    if (typeof(opts.glucose) !== 'undefined') {
+        //var glucoseData = opts.glucose;
+        var glucoseData = opts.glucose.map(function prepGlucose (obj) {
+            //Support the NS sgv field to avoid having to convert in a custom way
+            obj.glucose = obj.glucose || obj.sgv;
```

Comment:
Had to change this line to:
` obj.glucose = obj.glucose || obj.y;` 
to make this work for me. , The current logic doesn't seem to work for mmol. `sgv` is a mmol value at this point, transformed in the reporting code. (see my other comments on the pull request).

Reply:
@scottleibrand (or any other mgdl user) could you try to apply this change to see if it also works for you, I think that's the right fix.

Reply:
We'll need a more general solution, because autotune-prep/categorize.js is part of oref0, and has no concept of `y`.  I guess we'll need to look at how sgv is passed to this function, and pass `y` instead?

Reply:
Ah, that's right. Forgot about that. Yes.